test/claim-iat.test.js

```diff
@@ -53,22 +39,32 @@ describe('issue at', function() {
       {foo: 'bar'},
     ].forEach((iat) => {
       it(`should error with iat of ${util.inspect(iat)}`, function (done) {
-        expect(() => signWithIssueAtSync(iat, {})).to.throw('"iat" should be a number of seconds');
-        signWithIssueAtAsync(iat, {}, (err) => {
-          expect(err.message).to.equal('"iat" should be a number of seconds');
-          done();
+        signWithIssueAt(iat, {}, (err) => {
+          // the sign function swallows the error that is thrown by expect, so use done(err)
```

Comment:
Interesting, I didn't know about this, I guess it happens because of this: https://github.com/auth0/node-jsonwebtoken/blob/5498bdc4865ffb2ba2fd44d889fad7e83873bb33/sign.js#L176

We might tackle that at some point, for now, it's fine your handling here.

Reply:
Actually I just checked our current async tests, if they fail they fail with error:
`ensure "done()" is called` (because as you said sign function swallows the throw error.

Reply:
Yeah, I assumed the existing tests would have the same issue. This is definitely something I would like to fix eventually, assuming someone else doesn't get to it first.

Reply:
After looking into this in further detail, this would also be a problem is #511 was every addressed. 

Long term introducing Promises would be a possible solution as it would properly handle errors thrown in an asynchronous context.